MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm

```diff
@@ -693,6 +693,11 @@
 
 	encodeMetalArgumentBuffer(stage);
 
+	MVKPipeline* pipeline = getPipeline();
+	if (pipeline && pipeline->usesPhysicalStorageBufferAddressesCapability(stage)) {
+		getDevice()->encodeGPUAddressableBuffers(this, stage);
```

Comment:
I wonder if, eventually, we'll want to do some sort of GC-like tracing and only encode those buffers that are reachable from buffers that were directly passed--see (again) #1597 for why that's desirable. On the other hand, that might be even more difficult, given that it's difficult to tell if a write contained a buffer address...

In particular, I could see this horribly affecting performance in vkd3d, both due to encoding overhead and also due to memory pressure from keeping so many buffers in VRAM at once, since nearly every buffer it creates is going to have the `SHADER_DEVICE_ADDRESS` bit set.

Reply:
> GC-like tracing

A good idea, but I don't know if we could do that effectively. 

Since an app can pass an arbitrary `gpuAddress` to a shader, it can access any active `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT` buffer, which is what I assume here. 

Similarly, with descriptor-indexing, a shader could be commanded to arbitrarily access any of hundreds of thousands of resources in a huge descriptor-indexed Metal argument buffer, some based on indexes passed in via other buffers, some based on indexes derived from calculations within the shader itself.

But...this is indeed a real bottleneck issue with Metal, and I'm open to any suggestions! 

I was contemplating that `MTLHeaps` might help with streamlining encoding the resource usage, but it would be very hard to map arbitrary resources in an argument buffer to heaps that could be effectively encoded to make the contained resources resident.